engine_modules/market/tasks.py

```diff
@@ -1,8 +1,127 @@
 # -*- coding: utf-8 -*-
+from django.db.models import Max
+
 from engine.tasks import ResolutionTask
 from engine_modules.market.models import CorporationMarket
 
 
+class UpdateBubblesTask(ResolutionTask):
+	"""
+	Update the bubble value on the CorporationMarket objects
+	"""
+	RESOLUTION_ORDER = 500
+
+	def run(self, game):
+		# We can force the use of bubbles  using the force_bubbles flag
+
+		# I do NOT remember why the test was "not hasattr(game, 'force_bubbles')", I left it here just in case
+		# but the check == False is necessary. TODO: check if 'hasattr' has meaning (-> can a game not have an attribute 'force_bubbles' ?)
```

Comment:
yes it can.

In fact, the flag is either On or unset, that's why you don't need to check for False.